### PR TITLE
Added OpenAPI tool

### PR DESCRIPTION
**Pull request by mtirionMSFT**
This DocFxOpenAPI tool converts existing OpenAPI specification files into the format compatible with DocFX (OpenAPI v2 JSON files). It's now part of the docfx-companion-tools package. Please see the README in the source folder of the DocFxOpenAPI tool for more information on how to use it.

### CODE REVIEW

File: src/DocFxOpenApi/DocFxOpenApi.cs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+    /// <summary>
+    /// Open API file converter to V2 JSON files.
+    /// </summary>
+    internal class DocFxOpenApi
+    {
+        private const OpenApiSpecVersion OutputVersion = OpenApiSpecVersion.OpenApi2_0;
+        private static readonly string[] OpenApiFileExtensions = { "json", "yaml", "yml" };
+
+        private static int returnvalue;
```
Comment (Ellerbach) [inline src/DocFxOpenApi/DocFxOpenApi.cs:26]: maybe use the _something pattern for the global private variables?

File: src/DocFxOpenApi/Helpers/MessageHelper.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+{
+    using System;
+    using global::DocFxOpenApi.Domain;
+
+    /// <summary>
+    /// Helper methods to write messages to the console.
+    /// </summary>
+    public class MessageHelper
+    {
+        private readonly CommandlineOptions options;
```
Comment (Ellerbach) [inline src/DocFxOpenApi/Helpers/MessageHelper.cs:14]: same here _ pattern makes things easier to use (no need of the this. for example)

**Review by Ellerbach — COMMENTED** [review]
Couples of comments for the variable names. I know the rest of the code has already been reviewed

File: src/DocFxOpenApi/Helpers/MessageHelper.cs (hunk 2 above)
Comment (mtirionMSFT) [reply src/DocFxOpenApi/Helpers/MessageHelper.cs]: fixed

File: src/DocFxOpenApi/Helpers/MessageHelper.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        private readonly CommandlineOptions _options;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MessageHelper"/> class.
+        /// </summary>
+        /// <param name="options">Command line options.</param>
+        public MessageHelper(CommandlineOptions options)
+        {
+            this._options = options;
```
Comment (Ellerbach) [inline src/DocFxOpenApi/Helpers/MessageHelper.cs:22]: you should be able to remove all the `this.` in the file

File: src/DocFxOpenApi/DocFxOpenApi.cs [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    using Microsoft.OpenApi.Extensions;
+    using Microsoft.OpenApi.Models;
+    using Microsoft.OpenApi.Readers;
+
+    /// <summary>
+    /// Open API file converter to V2 JSON files.
+    /// </summary>
+    internal class DocFxOpenApi
+    {
+        private const OpenApiSpecVersion _outputVersion = OpenApiSpecVersion.OpenApi2_0;
```
Comment (Ellerbach) [inline src/DocFxOpenApi/DocFxOpenApi.cs:23]: keep the constants as they were but please change the private elements with the underscore pattern and remove the `this.` which are not necessary

**Pull request merged**